### ISSUE TITLE
Keep getting 199 Error

### ISSUE THREAD

**Issue by msaadbashir** [DigiX-Technologies/jazzcash_flutter#1]
Every time I do a test transaction it fails. When I go to sandbox API Call Log, it says 199 Error which according to documentation is System Error. Can you kindly highlight what might the issue be?

**Comment by ayaz-khan-niazi** [comment]
Did it work? I am going to implement it in my project. Please let me know if it worked for you.

**Comment by msaadbashir** [comment]
It hasn’t worked for me so far. If it works for you please do let me know

**Comment by ayaz-khan-niazi** [comment]
Sure. Thanks

**Comment by MrShafqatNadeem** [comment]
this package is built on the official documentation of the jazz cash, if its giving you some system errors then you should talk to the Jazz Cash officials, your credentials must have permissions to do a valid transaction.
or you can check the code status here -> https://sandbox.jazzcash.com.pk/SandboxDocumentation/v4.2/Resources.html

**Comment by msaadbashir** [comment]
But in the sandbox mode there shouldn't be any issue with regards to permission. I created a developer account on their website and tried using this resource based on the credentials generated via that.
As for reaching out to Jazz Cash officials, I haven't been able to do that using the form on their website. Not sure if there is any other method that I can use to reach out to them. 
Is this package currently working for you?

**Comment by MrShafqatNadeem** [comment]
yeah its working fine for me, i have tested it myself a moment ago just to confirm. and i have checked several apps as well in which  this package us being used, i think there must be something wrong with your credentials, you should talk to JAZZCASH MERCHANT SUPPORT (merchantsupport@jazzcash.com.pk)

**Comment by MrShafqatNadeem** [comment]
one of the app available is Imamia Jantri Official.
• Play Store : https://play.google.com/store/apps/details?id=com.iftikharbookdepot.imamia_jantri
• App Store : https://apps.apple.com/pk/app/imamia-jantri-official/id1441674194

**Comment by inzimambhatti** [comment]
@MrShafqatNadeem 
Can you please elaborate on return_url to me? The issue is that I am using my WordPress website as a return URL, and I am not getting any response from the method payViaJazzcash. Although my transaction was a success at the sandbox api call log.
I am getting the below exception while reading the Java script from the return URL.
 **"Uncaught TypeError: Cannot read properties of undefined (reading 'innerHTML')", source: https://my_return_url.com**
 your help well be very much appreciated.

**Comment by MrShafqatNadeem** [comment]
return url is something that gets the transaction data from the jazzcash payment api, 
so in detail you have to create an api or a page which accepts post request with payload. and it should return the same payload to the app to give you the proper response of the transaction.
here is the actual process ,

- a payment request is generated with a payload, 
- payload includes everything related to the product order , its price , jazzcash credentials(integrity salt, return url , password and everything else)  and its encrypted,
- request is sent to jazzcash payment api's with that encrypted payload.
- jazzcash returns a payment portal where we select our payment method , verifies captcha and proceeds the payment,
- once payment is done either via wallet or via some credit/debit card, it returns a json payload and visits that return url that you have provided in the payload at first. and your return url should behave like a reflector, it should return the same response as it was sent in request in json.
- and once that return url is visited the payment portal closed and the method startPayment() will return you that exact response and you can reckon and process that response to know if your payment is successful or not.


for return url :
for example create an api request with method POST , return the same payload that you are getting in the request.
you can write it in php if you have wordpress website. and you can process that payload on the website too if you want.

url : https://returnurl.com/payment_log
method : POST
request payload : (jsonObject)  => { 'key':'value' }
response body : (jsonObject)  => { 'key':'value' }

**Comment by inzimambhatti** [comment]
Mr.Shafqat Nadeem I have mailed you regarding the return url issue please, have a look and respond to it, as I am waiting for your response. Will be very thankful.

**Comment by msaadbashir** [comment]
A quick question, do you guys have just the sandbox account or have you registered yourselves as proper vendor before you can use Jazz Payment Gateway? Because I reached out to jazzcash merchant support ([merchantsupport@jazzcash.com.pk](mailto:merchantsupport@jazzcash.com.pk)) and they told me I need to provide them with necessary details of merchant registration before I can use sandbox mode even. Any clues?

**Comment by inzimambhatti** [comment]
Yes, we have an onboard JazzCash team with us. We have registered ourselves as merchants to Jazzcash. Both our live and test transactions are working fine. But what the issue I am facing during the integration is when I pay via credit or debit card via a JazzCash flutter package, I am not getting any successful or failure response from JazzCash. The main issue is my return URL JazzCash is not posting a response to my return URL. Although transactions succeed on API call log. But not getting a response after the transaction via card.
Saad Bashir is it working for you or not?

**Comment by msaadbashir** [comment]
Well being a freelancer I was hoping that it would work without me getting myself registered as a merchant so that I can build the product and later get the client registered directly. But reading your response it seems I will have to get myself registered as merchant too before I can get rid of this 199 Error which is rather unfortunate. I mean paypal and other international gateways don't have such restrictions on sandbox mode.

**Comment by inzimambhatti** [comment]
yeah I was facing 199 also, but once you registered yourself in jazzcash with your merchant account info your errors will be resolved. But not in live transactions. once Jazzcash make sure that your test transactions working fine. They will enable your live transactions in next step.

**Comment by inzimambhatti** [comment]
@MrShafqatNadeem Please help me with the return URL still waiting for the emails that I have sent you.

**Comment by TayyabAbdullah-9903** [comment]
> yeah I was facing 199 also, but once you registered yourself in jazzcash with your merchant account info your errors will be resolved. But not in live transactions. once Jazzcash make sure that your test transactions working fine. They will enable your live transactions in next step.

How did you solve this issue exactly? I am currently facing the same issue of receiving Code 199. Could you kindly guide me. Thank you!
